pulsar/schema.go

```diff
@@ -124,21 +125,10 @@ func NewSchema(schemaType SchemaType, schemaData []byte, properties map[string]s
 	return
 }
 
-type AvroCodec struct {
-	Codec *goavro.Codec
-}
-
-func NewSchemaDefinition(schema *goavro.Codec) *AvroCodec {
```

Comment:
This is a breaking change. Could we avoid it?

Is it possible to allow users to continue using goavro?

Reply:
we could, I guess :) -> but that would mean that there is dependency on both linkedin/goavro & hamba/avro 
using hamba internally and just have goavro.

personally I see no benefit of a function that does ... nothing

Reply:
@RobertIndie keeping `NewSchemaDefinition` function & `AvroCodec` struct with added Deprecation